src/Iesi.Collections/Generic/LinkedHashSet.cs

```diff
@@ -16,6 +16,9 @@ namespace Iesi.Collections.Generic
 	[Serializable]
 #endif
 	public class LinkedHashSet<T> : ISet<T>
+#if !NET40
```

Comment:
Instead of guarding NET 4.0 we could potentially add a reference to [ReadOnlyCollectionInterfaces](https://www.nuget.org/packages/ReadOnlyCollectionInterfaces/) package.

Reply:
Fine for me as done here, with guarding.